x/dex/keeper/msgserver/msg_server_contract_deposit_rent.go

```diff
@@ -36,12 +36,18 @@ func (k msgServer) ContractDepositRent(goCtx context.Context, msg *types.MsgCont
 		// a sender can only "claim" the contract if the rent balance is 0
 		return nil, sdkerrors.ErrUnauthorized
 	}
-	// deposit
-	creatorAddr, err := sdk.AccAddressFromBech32(contract.Creator)
+
+	senderAddr, err := sdk.AccAddressFromBech32(contract.Creator)
+	// claim a contract when the contract.RentBalance==0 should be executed from msg.Sender
+	if contract.RentBalance == 0 {
+		senderAddr, err = sdk.AccAddressFromBech32(msg.Sender)
+	}
 	if err != nil {
 		return nil, err
 	}
-	if err := k.BankKeeper.SendCoins(ctx, creatorAddr, k.AccountKeeper.GetModuleAddress(types.ModuleName), sdk.NewCoins(sdk.NewCoin(appparams.BaseCoinUnit, sdk.NewInt(int64(msg.Amount))))); err != nil {
+
+	// deposit
+	if err := k.BankKeeper.SendCoins(ctx, senderAddr, k.AccountKeeper.GetModuleAddress(types.ModuleName), sdk.NewCoins(sdk.NewCoin(appparams.BaseCoinUnit, sdk.NewInt(int64(msg.Amount))))); err != nil {
```

Comment:
doesnt this make it such that anybody could force the creator account to transfer funds to the contract rent (if rent != 0). Isn't this an issue since rent can't be withdrawn so anyone could force the account to have its whole balance to be locked in the rent (almost like burning?)